Refactor(eos_cli_config_gen, eos_designs): Improve BGP VRF Address Families Model (#2808)
aristanetworks/ansible-avd: ansible_collections/arista/avd/docs/porting-guides/4.x.x.md

              - send any
          neighbors:
            - ip_address: 10.2.3.4
              activate: true
              route_map_out: RM-10.2.3.4-SET-NEXT-HOP-OUT
          networks:
            - prefix: 100.64.0.0/10
              route_map: RM-10.2.3.4
          peer_groups:
            - name: TEST_PEER_GRP
              activate: true
              next_hop:
                address_family_ipv6_originate: true
        - address_family: ipv6
          ...
```

Example with new data model (discrete dictionary keys for all supported address families):

```yaml
router_bgp:
  vrfs:
    - name: VRF001
      address_family_ipv4:
        bgp:
          additional_paths:
            send:
              any: true
        neighbors:
          - ip_address: 10.2.3.4
            activate: true
            route_map_out: RM-10.2.3.4-SET-NEXT-HOP-OUT
        networks:
          - prefix: 100.64.0.0/10
            route_map: RM-10.2.3.4
        peer_groups:
          - name: TEST_PEER_GRP
            activate: true
            next_hop:
              address_family_ipv6:
                enabled: true
                originate: true
      address_family_ipv6:
        ...
```

> The change has been incorporated into `eos_designs` so no action is necessary to retain the same functionality. If using `structured_configuration` to add additional configuration options under the address families using the old data models, migrating to using the new data models is recommended.

Reviewer: ```suggestion
The change has been incorporated into `eos_designs` so no action is necessary to retain the same functionality.

It is recommended to migrate to the new data models when using `structured_config` or `custom_structured_configuration_` to add additional configuration options under the VRF address families.
```
